api/app/Http/Controllers/EntityController.php

```diff
@@ -61,7 +61,12 @@ public function getAllPaginated(Request $request, RangeRequest $rangeRequest)
         $offset = $rangeRequest->isGetLast() ? $totalCount - $limit : $rangeRequest->getOffset();
 
         if ($request->has('q')) {
-            $collection = $this->searchAllEntities($request->query('q'), $limit, $offset, $totalCount);
+            $queryArray = json_decode($request->query('q'), true);
```

Comment:
I think all this logic would be better suited outside of this method, so that just the `$request->query('q')` is passed, then the simple/complex decision is made. Just thinking about controllers that extend this method for reasons other than querying; they shouldn't have to repeat this logic.